File: testing/test-data/c-sharp/password_checker.cs
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science 
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Microsoft Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace AdaCodeSamples {
    class TestData {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            string passwordToCheck = "test";
            bool result = CheckPassword(passwordToCheck);
            Console.WriteLine($"Valid password? {result}");
        }
        

        // Checks that a password length is between 8 and 14 characters. Returns True if valid
        public static bool CheckPassword(string password) {
            bool isValid = true;
            
            if (password.Length < 8 || password.Length > 14) {
                isValid = false;
            }

            return isValid;
        }


    }
}

File: testing/test-data/password_checker.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-SA 4
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program

    Sub Main()
        Dim passwordToCheck As String = "test"
        Dim result As Boolean = CheckPassword(passwordToCheck)
        Console.WriteLine(result)

    End Sub

    Function CheckPassword(ByVal password As String) As Boolean
        Dim isValid As Boolean = True

        If password.Length < 8 OrElse password.Length > 14 Then
            isValid = False
        End If

        Return isValid
    End Function



End Module
